### PR TITLE
Matrix export

### PR DESCRIPTION
**Pull request by pedrocamargo**
Addresses matrix export issue reported on the GUI repo.

https://github.com/AequilibraE/AequilibraE-GUI/issues/171

### CODE REVIEW

File: aequilibrae/matrix/aequilibrae_matrix.py [hunk 1]
```diff
@@ -702,26 +704,16 @@ def export(self, output_name: str, cores: List[str] = None):
...
-            self.computational_view(names)
+            df = None
+            for core in self.names:
+                coo = coo_matrix(self.matrix[core])
+                df2 = pd.DataFrame({"row": self.index[coo.row], "column": self.index[coo.col], core: coo.data})
+                df2.set_index(["row", "column"], inplace=True)
+                if df is None:
+                    df = df2
+                else:
+                    df = df.join(df2, how="outer")
```
Comment (jamiecook) [inline aequilibrae/matrix/aequilibrae_matrix.py:715]:
```suggestion
            def f(name): 
                coo = coo_matrix(self.matrix[name])
                data = {"row": self.index[coo.row], "column": self.index[coo.col], core: coo.data}
                return pd.DataFrame(data).set_index(["row", "column"])
                
            dfs = [f(name) for name in self.names]
            df = reduce(lambda a,b: a.join(b,how='outer'), dfs)     
```

File: tests/aequilibrae/project/test_matrices.py [hunk 2]
```diff
@@ -29,90 +30,90 @@ def tearDown(self) -> None:
...
-        self.assertEqual(df.shape[0], self.curr.fetchone()[0], 'Returned the wrong number of matrices in the database')
+        self.curr.execute("select count(*) from Matrices")
+        self.assertEqual(df.shape[0], self.curr.fetchone()[0], "Returned the wrong number of matrices in the database")
 
-        self.assertEqual(df[df.status == 'file missing'].shape[0], 1, 'Wrong # of records for missing matrix files')
+        self.assertEqual(df[df.status == "file missing"].shape[0], 1, "Wrong # of records for missing matrix files")
 
     def __mat_count(self, should_have: int, error_message: str) -> None:
-        self.curr.execute('Select count(*) from Matrices;')
+        self.curr.execute("Select count(*) from Matrices;")
```
Comment (jamiecook) [inline tests/aequilibrae/project/test_matrices.py:118]: You should do one PR where you just run black over the entire codebase and then never worry about " vs ' again

Comment (pedrocamargo) [reply tests/aequilibrae/project/test_matrices.py:118]: Will do

File: aequilibrae/matrix/aequilibrae_matrix.py (hunk 1 above)
Comment (pedrocamargo) [reply aequilibrae/matrix/aequilibrae_matrix.py]: Good one

**Comment by janzill** [comment]
@pedrocamargo part of this is about internal handling of matrix names, right? I think making comparisons case-insensitive is a good idea, just needs a warning/note in the documentation. We might want to normalise things at some point to deal with non-ASCII letters, see for example https://stackoverflow.com/a/29247821 but I think this is good for now

**Review by janzill — APPROVED** [review]

**Pull request merged**